src/code/z_debug_mode.c

```diff
@@ -1,13 +1,160 @@
+#include "z64debug.h"
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_debug_mode/func_800E9C90.s")
+typedef struct {
+    /* 0x0 */ u8 colorIndex;
+    /* 0x1 */ char text[11];
+} DebugTextBufferEntry; // size = 0xC
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_debug_mode/func_800E9CA0.s")
+typedef struct {
+    /* 0x0 */ u8 x;
+    /* 0x1 */ u8 y;
+    /* 0x2 */ u8 colorIndex;
+    /* 0x3 */ char text[41];
+} DebugCamTextBufferEntry; // size = 0x2C
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_debug_mode/func_800E9CFC.s")
+#define DEBUG_TEXT_DRAW_TEXT (1 << 0)
+#define DEBUG_TEXT_DRAW_CAM_TEXT (1 << 1)
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_debug_mode/func_800E9DBC.s")
+s32 sDebugTextDrawFlags = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_debug_mode/func_800E9E94.s")
+DebugTextBufferEntry sDebugTextBuffer[] = {
+    { 0, "          " }, { 0, "          " }, { 0, "          " },
```

Comment:
```suggestion
DebugTextBufferEntry sDebugTextBuffer[] = {
    { DEBUG_TEXT_BLACK, "          " }, { 0, "          " }, { 0, "          " },
```
Is it worth filing these in since they get overwritten anyway?

Reply:
Seems like a good reminder that that struct member is the color index.